### ISSUE TITLE
GitHub Enterprise Server Support

### ISSUE THREAD

**Issue by wreiske** [RocketChat/Apps.Github22#111]
Please add support for self-hosted GitHub Enterprise installs. 

This _should_ be as simple as allowing the user to configure the API endpoint URL in the settings, rather than just defaulting to api.github.com. 

You can setup a free 45 day trial on enterprise server for testing, https://docs.github.com/en/get-started/signing-up-for-github/setting-up-a-trial-of-github-enterprise-server

**Comment by VipinDevelops** [comment]
Hey @wreiske, I'm extremely sorry for the late reply. Thank you so much for providing reviews on our app and suggesting new enhancements. Do you still need this feature? If yes, I can start working on it right away.

**Comment by wreiske** [comment]
Yes, please!! It would be great to use this app with GitHub Enterprise Server on-prem.

**Comment by VipinDevelops** [comment]
Certainly! Thank you for your suggestion @wreiske . We will make sure the development and deployment of this feature as quickly as possible. Do you have any more suggestions or ideas to enhance the app?

**Comment by VipinDevelops** [comment]
Hey @wreiske  I've finished building the feature, but I'm stuck without a testing environment. Any chance you could hook me up with access to a server or a free provider for a quick test spin?

**Comment by TobiasMeyster** [comment]
Glad to hear that this feature will be coming :)

As he mentioned in his first post u could use a 30 day free trial for Github Enterprise Cloud here https://docs.github.com/en/enterprise-cloud@latest/admin/overview/setting-up-a-trial-of-github-enterprise-cloud

I think for most large companies that will be a very helpfull enhancement as most of them own an self hosted or cloud instance of Github due to data security :)

**Comment by VipinDevelops** [comment]
@TobiasMeyster, @wreiske, thank you for your patience. @samad-yar-khan and I had a brief discussion about how we can address the issue. Although the problem may seem small, there are numerous moving parts in the app that might be affected. We are actively working on resolving this issue, so I kindly ask for a bit more patience. Once the code part is completed, we will create a pull request for you to test it out. Unfortunately, we don't have access to a server to host a GitHub Enterprise, and we hope you understand.
